hilti/runtime/src/library.cc

```diff
@@ -130,5 +131,18 @@ hilti::rt::Result<hilti::rt::Nothing> hilti::rt::Library::save(const hilti::rt::
     if ( ec )
         return result::Error(fmt("could not save library to %s: %s", path, ec.message()));
 
+    // Query the current umask. This is safe since we always query and set the umask from a single thread.
+    auto default_perms = ::umask(::mode_t());
+    ::umask(default_perms);
+
+    // Create the file taking into account the active umask. We only add
+    // permissions and never remove masked ones since the user might have set
+    // one which masks required permissions.
+    hilti::rt::filesystem::permissions(path, hilti::rt::filesystem::perms(default_perms),
+                                       hilti::rt::filesystem::perm_options::add, ec);
```

Comment:
Let's say our umask is `022`. Without this patch, the ticket says that `path` is created with `-rwx--x--x`. Wouldn't this then add write permissions for group/other to that, so we'd end up with `-rwx-wx-wx`. But what we'd want with that umask is `rwxr-xr-x`. What am I missing?

Reply:
Fixed as discussed offline.